### ISSUE TITLE
ejabberd odbc error in module ejabberd_auth_odbc

### ISSUE THREAD

**Issue by harshsuri** [processone/ejabberd#796]
I have the following configuration
* Ubuntu 14.04 *(OS)*
* erl/otp 18.1
* ejabberd 15.09.48
* UnixODBC 2.3.4 *(ODBC)*
* FreeTDS 0.95 *(driver)*
* Azure SQL *(backend)*

Situation: 
1. `osql` test works fine
2. odbc connection and queries run fine in the `erl` shell
3. odbc connection and queries run fine in the `ejabberdctl live` mode

But for a running server, I get the following error if I try to connect any user from a client:

```ejabberd_auth:is_user_exists:316 The authentication module ejabberd_auth_odbc returned an error
when checking user <<"xxxxx">> in server <<"xmpp.xxxx.xxx">>
Error message: {aborted,{no_exists,[sql_pool,<<"xmpp.xxxx.xxx">>]}}```

Any ideas?  What am I doing wrong or is it a bug somewhere?

I configured the ejabberd using `./configure --enable-odbc`

**Comment by harshsuri** [comment]
Not sure how the reference appeared but that was not the intent.  Please ignore the referenced issue.

**Comment by zinid** [comment]
This message typically arises when the virtual host is unknown, i.e. you do not have "xmpp.xxxx.xxx" configured or you didn't configure odbc for this vhost.

**Comment by harshsuri** [comment]
No,  that is just a placeholder.  The configuration  works perfectly if I test is using osql, or even  via the erl shell using the odbc commands.  Further, if I start the ejabberd server  in  live mode, I enter the shell and connect to the server and run queries.  I receive the data.  The only issue is when I keep the server on for a few hours and it disconnect and do not re-connect.  I have  tired using various inputs for keepalive parameter but nothing seems to work.  After a few hours, the ejabberd server  will not be  able to  connect to the sql sever in the backend.  I tried many ways and after struggling for a week or so, I figured let me post a question.
